drivers/volume/portworx/portworx.go

```diff
@@ -642,22 +641,40 @@ func (d *portworx) GetNodeForVolume(vol *torpedovolume.Volume) (*node.Node, erro
 		}
 	}
 
-	pxVol := v.(*api.Volume)
-	for _, n := range node.GetStorageDriverNodes() {
-		if n.VolDriverNodeID == pxVol.AttachedOn {
-			return &n, nil
+	r := func() (interface{}, bool, error) {
+		pxVol := v.(*api.Volume)
+		for _, n := range node.GetStorageDriverNodes() {
+			if n.VolDriverNodeID == pxVol.AttachedOn {
+				return &n, false, nil
+			}
+		}
+
+		// Snapshots may not be attached to a node
+		if pxVol.Source.Parent != "" {
+			logrus.Debugf("Volume with ID: %s is a snapshot", pxVol.Id)
+			return nil, false, nil
+		}
+
+		return nil, true, &ErrFailedToInspectVolume{
+			ID:    name,
+			Cause: "Volume is not attached on any node",
 		}
 	}
 
-	// Snapshots may not be attached to a node
-	if pxVol.Source.Parent != "" {
-		return nil, nil
+	n, err := task.DoRetryWithTimeout(r, defaultTimeout, defaultRetryInterval)
```

Comment:
Use a 30 second timeout instead of the default 2 minutes.

Reply:
Made custom timeout vars for validating volume attached